core/src/main/java/be/xplore/pricescraper/domain/users/User.java

```diff
@@ -0,0 +1,66 @@
+package be.xplore.pricescraper.domain.users;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import java.util.Collection;
+import java.util.List;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+/**
+ * The User of our application.
+ */
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+public class User implements UserDetails {
```

Comment:
The domain should have as little dependencies as possible. Especially large dependencies as spring security are better implemented elsewhere. It's a good thing to have your own concept of a user in the domain, but it should not be influenced by external dependencies.

Remember: hexagonal architecture intends to keep the domain clean, and free of implementation details. I would cal spring security an implementation detail.

Reply:
it depends on what the team agrees on, some compromises could be made to have like spring-security in the core